CHANGELOG.md

```diff
@@ -0,0 +1,30 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+Changelog was added with version 2.0.0.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [2.0.0] - 2021-05-05
+
+Python 2 is no longer supported. This is a maintenance release without
+that many added features. It was mostly done to make future development easier.
+
+### Added
+- Storage upload accepts filenames in strings and PathLike or BinaryIO variables.
+- Code style is now guarded by Black, flake8, isort etc.
+- Improved documentation and its examples, especially regarding server creation and storage uploads.
+
+### Changed
+- Huge amount of fixups in project tests, style and imports by [akx](https://github.com/akx). Thank you! :heart:
+- Zone default from storage creation has been removed, making zone a required variable with `create_storage()`.
+- Passwords for server user are not created by default if SSH keys are provided.
+- Tests and deployments moved fully from CircleCI to GitHub Actions.
+- Fixed storage upload not reading the file into memory before uploading.
+- Moved to fully using setup.cfg instead of requirements.txt.
+
+### Removed
+- Python 2 support.
```

Comment:
Hmm.... Should we include the old versions here as well?

Reply:
Something like:

v1.0.1
11 Feb 2021

Fixed

long description for PyPi


v1.0.0
11 Feb 2021

Added

Features: Object Storage, Network, router, hosts
user-agent to follow usage trend
Storage methods clone #6 , templatize, import, favorite and backup #5
tests
Changed

Breaking: Update to UpCloud API 1.3
Breaking: Update supported Python versions
Python 2.7 and Python >3.6
Breaking: Update dependencies
Move from travis to github actions
Fixed

Fetch Zones and templates from API
Refactoring
Update Readme


v0.4.6
28 Aug 2020

Support for new templates: CentOS 8.0 and Ubuntu 20.04
Removed obsolete templates
Added US-NYC1 zone constant


v0.4.5
3 Dec 2019

Template updates.


v0.4.3
3 Dec 2018

Python 3.3 deprecated
Zone San Jose added
Template Ubuntu 18.04 added


v0.4.2
22 Feb 2018

Added missing zones and missing templates.

Also removed old template windows 2008, which is not supported any more.


v0.4.1
2 Feb 2018

CI fixes
Small bug fix


v0.4.0
1 Feb 2018 


v0.3.9
16 Nov 2016
Note: 0.3.8 was never released at pip / github releases.

numerous improvements from the 0.4.0 branch #29
support for creating a server from template
https://www.upcloud.com/api/8-servers/#creating-from-a-template
``` python
manager.create_server(
    Server(
        core_number=2,
        memory_amount=1024,
        hostname='my.example.com',
        zone=ZONE.Frankfurt,
        storage_devices=[
            Storage(uuid=UUID, size=10),  # we also support Storage(storage=UUID, size=10)
        ],
    )
)
```


v0.3.7
4 Aug 2016

support for Server.user_data
improve Server.stop_and_destroy
automatically populates server data from api unless sync=False param is given
wraps stop and destroy operations in an error handling loop that tries the operations several times in case of API errors related to the resource being in "maintenance" state (servers sometimes go to "maintenance" state when destroying several servers in a row)

Reply:
Let's add them after releasing this one.